Libs/Python/ShapeworksPython.cpp

```diff
@@ -1116,10 +1116,60 @@ PYBIND11_MODULE(shapeworks_py, m)
        "point"_a)
 
   .def("geodesicDistance",
-       &Mesh::geodesicDistance,
+       py::overload_cast<int, int>(&Mesh::geodesicDistance),
        "computes geodesic distance between two vertices (specified by their indices) on mesh",
        "source"_a, "target"_a)
 
+  .def("geodesicDistance",
+       [](Mesh &mesh, const std::vector<double> p) -> decltype(auto) {
+          auto array = mesh.geodesicDistance(Point({p[0], p[1], p[2]}));
+          const auto shape = std::vector<size_t>{static_cast<unsigned long>(array->GetNumberOfTuples()),
+                                                 static_cast<unsigned long>(array->GetNumberOfComponents()),
+                                                 1};
+          auto vtkarr = vtkSmartPointer<vtkDoubleArray>(vtkDoubleArray::New());
+          vtkarr->SetNumberOfValues(array->GetNumberOfValues());
+
+          // LOTS of copying going on here, see github #903
+          array->GetData(0, array->GetNumberOfTuples()-1,
+                         0, array->GetNumberOfComponents()-1,
+                         vtkarr);                               // copy1
+          return py::array(py::dtype::of<double>(),
+                         shape,
+                         vtkarr->GetVoidPointer(0));          // copy2
+       },
+       "computes geodesic distance between a point (landmark) and each vertex on mesh",
+       "landmark"_a)
+
+  .def("geodesicDistance",
+       [](Mesh &mesh, const std::vector<std::vector<double>> p) -> decltype(auto) {
+          std::vector<Point> points;
+          for (int i=0; i<p.size(); i++)
+          {
+            points.push_back(Point({p[i][0], p[i][0], p[i][2]}));
+          }
+          auto array = mesh.geodesicDistance(points);
```

Comment:
Please create a function for this section (down to line 1163) since it's identical to the lambda above:
`py::array vtkToNumpy(vtkArray)`. 

In fact, I suggest `vtkToEigen` since what we have is a nComponents x nPoints matrix, i.e., `Eigen::MatrixXd(double, nComponents, nPoints)`, because it's actually much simpler to write and pybind11 takes care of moving it back and forth. I just implemented it for itk arrays in EigenUtils.h/cpp in the as-yet-unpushed branch that fixes Transforms. I made a simple copy version for that, and if you do that you can still reference the "don't copy arrays passed between c++ and Python" issue above :). 

The more appropriate version should use `Eigen::Map<...>` [as described here](https://eigen.tuxfamily.org/dox/group__TutorialMapClass.html), and as used in the other functions in EigenUtils.h. 

I'm copying this comment to the issue #903 if you just want to make a simple wrapper of this duplicated code in these two lambdas for now since it works.

Reply:
Function created, and lots of duplicated code consolidated.